src/js/stores/DCOSStore.js

```diff
@@ -79,6 +89,24 @@ class DCOSStore extends EventEmitter {
     MarathonStore.fetchServiceVersions(serviceID);
   }
 
+  onMarathonDeploymentsChange() {
+    let deploymentsList = MarathonStore.get('deployments');
+    let serviceTree = MarathonStore.get('groups');
+
+    // Populate deployments wth affected services
+    this.data.marathon.deploymentsList = deploymentsList
+      .mapItems(function (deployment) {
+        let ids = deployment.getAffectedServiceIds();
+        let services = ids.map(function (id) {
```

Comment:
Could shorten this to just `ids.map(serviceTree.findItemById);` as an alternative :)